### ISSUE TITLE
False positive E0594

### ISSUE THREAD

**Issue by busyluo** [intellij-rust/intellij-rust#4819]
## Environment

* **IntelliJ Rust plugin version:** 0.2.114.2822-193-nightly
* **Rust toolchain version:** 1.40.0-stable-gnu
* **IDE name and version:** CLion 2019.3.1
* **Operating system:** Windows 10 1903

## Problem description
![image](https://user-images.githubusercontent.com/16347811/71957419-afef5600-3228-11ea-993c-6ef0101aa87c.png)

## Steps to reproduce
``` rust
fn foo() {
    let mut v = b"abc".clone();
    v[0] = 1;
}
```
